master/internal/db/setup.go

```diff
@@ -83,24 +85,64 @@ func Connect(opts *config.DBConfig) (*PgDB, error) {
 	return db, nil
 }
 
+// IsNew checks to see if the database's migration tracking tables have been
+// created, and if so, if it's above version 0. It returns `false` if the current
+// version exists and is higher than zero, `true` otherwise.
+// This is not guaranteed to be accurate if the database is otherwise in a bad or
+// incomplete state. If an error is returned, the bool should be ignored.
+func IsNew(opts *config.DBConfig) (bool, error) {
```

Comment:
we should add a test for this. i'd prefer a test that runs core.go::master.Run and tests the flow of creating a cluster many times without a password and the failure not going away. but testing this in isolation might make sense too.

Reply:
Added a test; unfortunately running said test with a good-enough password allowed it to get all the way to running the service, which took me down a bit of a rabbit-hole figuring out how to shut the thing down.

Reply:
yeah, went i got to do the "removed actors from the agent" spike i had enough time to add graceful shutdown via context cancellation. master hasn't been so fortunate.

Reply:
oh jk, i think i (or someone) did some work towards this: https://github.com/determined-ai/determined/blame/main/master/internal/core.go#L1123. i'd be interested to learn more about what went south.